modules/33acrossBidAdapter.js

```diff
@@ -208,31 +198,103 @@ function _createServerRequest({bidRequest, gdprConsent = {}, uspConsent, pageUrl
   return {
     'method': 'POST',
     'url': url,
-    'data': JSON.stringify(contributeViewability(ttxRequest)),
+    'data': JSON.stringify(ttxRequest),
     'options': options
   }
 }
 
-// **************************** BUILD REQUESTS: BIDFLOORS *************************** //
-function _getBidFloors(getFloor, size) {
+// BUILD REQUESTS: SIZE INFERENCE
+function _transformSizes(sizes) {
+  if (utils.isArray(sizes) && sizes.length === 2 && !utils.isArray(sizes[0])) {
+    return [ _getSize(sizes) ];
+  }
+
+  return sizes.map(_getSize);
+}
+
+function _getSize(size) {
+  return {
+    w: parseInt(size[0], 10),
+    h: parseInt(size[1], 10)
+  }
+}
+
+// BUILD REQUESTS: PRODUCT INFERENCE
+function _getProduct(bidRequest) {
+  const { params, mediaTypes } = bidRequest;
+  const videoAdUnit = utils.deepAccess(bidRequest, 'mediaTypes.video', {});
+  const { context } = videoAdUnit;
+
+  const { banner, video } = mediaTypes;
+
+  if ((video && !banner) && context === 'instream') {
+    return PRODUCT.INSTREAM;
+  }
+
+  return (params.productId === PRODUCT.INVIEW) ? (params.productId) : PRODUCT.SIAB;
+}
+
+// BUILD REQUESTS: BANNER
+function _buildBannerORTB(bidRequest) {
+  const bannerAdUnit = utils.deepAccess(bidRequest, 'mediaTypes.banner', {});
+  const element = _getAdSlotHTMLElement(bidRequest.adUnitCode);
+
+  const sizes = _transformSizes(bannerAdUnit.sizes);
+
+  let format;
+
+  // We support size based bidfloors so obtain one if there's a rule associated
+  if (typeof bidRequest.getFloor === 'function') {
+    let getFloor = bidRequest.getFloor.bind(bidRequest);
+
+    format = sizes.map((size) => {
+      const bidfloors = _getBidFloors(getFloor, size, MEDIA_TYPE.BANNER);
+
+      let formatExt;
+      if (bidfloors) {
+        formatExt = {
+          ext: {
+            ttx: {
+              bidfloors: [ bidfloors ]
+            }
+          }
+        }
+      }
+
+      return Object.assign({}, size, formatExt);
+    });
+  } else {
+    format = sizes;
+  }
+
+  const minSize = _getMinSize(sizes);
+
+  const viewabilityAmount = _isViewabilityMeasurable(element)
+    ? _getViewability(element, utils.getWindowTop(), minSize)
+    : NON_MEASURABLE;
+
+  const contributeViewability = ViewabilityContributor(viewabilityAmount);
+
+  return {
+    format,
+    contributeViewability
```

Comment:
with the change in `ViewabilityContributor`, maybe remove the ViewabilityContributor and return the ext object in here `viewability: contributeViewability(viewabilityAmount)`